File: stage/src/main/java/com/matcheval/stage/model/Candidature.java
package com.matcheval.stage.model;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

@Entity
@Data
public class Candidature {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private LocalDateTime dateSoumission;
    private String statut;
    private String commentaire;

    private String candidatNom;
    private String candidatPrenom;
    private String candidatEmail;

    @ManyToOne
    private OffreEmploi offre;
    @OneToOne(mappedBy = "candidature", cascade = CascadeType.ALL)
    private CV cv;

}

File: coreui-free-angular-admin-template-main/src/app/models/Candidature.ts
// src/app/models/candidature.model.ts
export interface candidature {
  id: number;
  dateSoumission: string;
  statut: string;
  commentaire: string;
  candidatNom: string;
  candidatPrenom: string;
  candidatEmail: string;
  cvOriginalFilename?: string;
  hasCvText?: boolean;
  offreId?: number;
  offreTitre?: string;
}